### DISCUSSION TITLE
SD Compatibility

### DISCUSSION THREAD

**Discussion by sanni** [sanni/cartreader#206] [General Chat]
### This is a list of tested cards that work with Cart Reader version 6.4 and up.

To test an SD card please use the Windows formatting tool(This PC->Right Click on SD card -> Format) and choose exFAT and default allocation size.    
Then [copy the SD files](https://github.com/sanni/cartreader/wiki/User-Guide) to the root of the SD card and dump an N64 game cartridge. The SD card is considered working when the checksum at the end of the dump is correct.

| Manufacturer  | Speed    | Size | Format | Works | Picture |
|---------------|----------|------|--------|------|---------|
| Sandisk Ultra | Class 10 | 64GB | exFAT  | Yes  | ![sd64gb](https://user-images.githubusercontent.com/19169494/126880410-099a9d74-1aa8-4ac1-8d5a-d8834b767ed8.jpg) |
| no name       | Class 6  | 8GB  | exFAT  | Yes  | ![no8](https://user-images.githubusercontent.com/19169494/126880408-55112afd-eba4-44a6-b48f-007d1c73712b.jpg)    |
| Sandisk       | Class 4  | 8GB  | exFAT  | Yes  | ![sd8gb](https://user-images.githubusercontent.com/19169494/126880409-03e12021-ad6e-46cf-96ce-9d3867c0f8bd.jpg)  |
| no name       | /        | 1GB  | exFAT  | Yes  | ![no1](https://user-images.githubusercontent.com/19169494/126880407-36318baa-49c7-400c-9b14-b46e838c3f53.JPEG)   |

You can also post here if your SD card does not work.

**Comment by PsyK0p4T** [comment]
<html>
<body>


Manufacturer | Speed | Size | Format | Works | Picture
-- | -- | -- | -- | -- | --
onn. | Class 10 U1 | 32GB | exFAT | Yes |  ![onn](https://user-images.githubusercontent.com/87064902/126899433-d1b7ab0d-8551-4baf-a87a-35dd66fe1c45.jpg)


</body>
</html>

**Comment by Dakkaron** [comment]
Pleas don't take this the wrong way. Is there any point to the benchmark part?

According to Wikipedia, the slowest SD cards from 2001 had a write speed of 6.4 MBit/s and a read speed of 28.8 MBit/s.
The Arduino Mega has a default clock frequency of 16 MHz. So reading-wise the first ever SD card would outclass the theoretical maximum bandwidth an Arduino Mega is serially able to perform. And in terms of write-speed, once you factor in that the Mega would need to get data from somewhere or do any kind of processing (even if it is only switching the clock signal high/low), the Mega would be very hard pressed to outperform the SD.

And we are talking about an SD from 2001.

A class 2 SD card (which is the lowerst speed grade to have a name) has 16 MBit/s bandwidth and will outperform the ATMega in any circumstance. So the time limit of 21sec is just benchmarking the ATMega, not the SD.

You could try to overclock the Mega though and see if that changes anything.

**Reply by sanni** [reply]
You are right. I changed the "Time" column to "Works", since the time will always be the same anyway.

**Comment by StanMOI** [comment]
<!DOCTYPE html>

Manufacturer | Speed | Size | Format | Works | Picture
-- | -- | -- | -- | -- | --
? | Class 10 U1 | 8GB | FAT32 & exFAT | Yes |  ![onn](https://user-images.githubusercontent.com/38383423/128215365-8fbbf979-88ba-4066-970a-cee128b5f0ad.png)
